packages/esm-patient-list-app/src/api/hooks.ts

```diff
@@ -30,7 +30,10 @@ export function useAllPatientListsWhichDoNotIncludeGivenPatient(patientUuid: str
 }
 
 export function usePatientListDetails(patientListUuid: string) {
-  const swrResult = useSWR<FetchResponse<OpenmrsCohort>, Error>(`${cohortUrl}/cohort/${patientListUuid}`, openmrsFetch);
+  const swrResult = useSWR<FetchResponse<OpenmrsCohort>, Error>(
+    `${cohortUrl}/cohort/${patientListUuid}?v=custom:(uuid,name,description,display,size,attributes,startDate,endDate)`,
+    openmrsFetch,
+  );
   return { ...swrResult, data: swrResult?.data?.data };
 }
```

Comment:
```suggestion
  const url = `${cohortUrl}/cohort/${patientListUuid}?v=custom:(uuid,name,description,display,size,attributes,startDate,endDate)`;

  const { data, error, isLoading, mutate } = useSWR<FetchResponse<OpenmrsCohort>, Error>(
    patientListUuid ? url : null,
    openmrsFetch,
  );

  return {
    data: data?.data,
    error,
    isLoading,
    mutate,
  };
```

Reply:
This is the established pattern for fetching data using SWR hooks. The API URL gets extracted into a variable so that it's easy to make changes to it without affecting the readability of the `useSWR` hook invocation. The `data`, `error`, `isLoading`, and `mutate` properties get destructured from the hook. On line 36, the `url` gets supplied to the hook only if a truthy `patientListUuid` value is provided. This ensures an API request doesn't get made without a `patientListUuid`. Finally, the hook returns the destructured properties.